proto/ord_schema.proto

```diff
@@ -40,73 +40,174 @@ message Reaction {
 message ReactionIdentifier {
   // Possible identifier types are listed in an enum for extensibility
   enum IdentifierType {
-    REACTION_IDENTIFIER_TYPE_UNKNOWN = 0;  
+    REACTION_IDENTIFIER_TYPE_UNKNOWN = 0;  // Unspecified or custom.
     REACTION_IDENTIFIER_TYPE_REACTION_SMILES = 1;  // Reaction SMILES.
     REACTION_IDENTIFIER_TYPE_ATOM_MAPPED_SMILES = 2;  // Atom-mapped SMILES.
     REACTION_IDENTIFIER_TYPE_RINCHI = 3;  // Reaction InChI.
     REACTION_IDENTIFIER_TYPE_NAME = 4;  // Named reaction or reaction category.
   }
   IdentifierType type = 1;
-  // Custom reaction identifier 
+  // Custom reaction identifier, used only when 'type' field is UNKNOWN.
   string type_custom = 2;
   string value = 3;
 }
 
 message ReactionInput {
   Compound compound = 1;
   enum ReactionRole {
-    REACTION_ROLE_UNKNOWN = 0;
+    // Unspecified.
+    REACTION_ROLE_UNKNOWN = 0;  
+    // TODO(ccoley): Do we want to use the definition of a reactant aligned
+    // with Reaxys, or say that any species that contributes heavy atoms 
+    // to a desired product is a reactant? This field might be kind of a 
+    // throwaway anyway...
     REACTION_ROLE_REACTANT = 1;
     REACTION_ROLE_REAGENT = 2;
     REACTION_ROLE_SOLVENT = 3;
     REACTION_ROLE_CATALYST = 4;
   }
   ReactionRole reaction_role = 2;
-  // Whether this species should be considered a limiting reactant.
+  // Whether this species was intended to be a limiting reactant.
   bool is_limiting = 3;
   /**
    * Used to define order of addition. ReactionInputs with the same 
-   * addition_stage were added simultaneously. One ReactionInput with a 
-   *  lower addition_stage than another was added earlier in the procedure.
+   * addition_order were added simultaneously. One ReactionInput with a 
+   * lower addition_order than another was added earlier in the procedure.
+   * This field is 1-indexed.
    */
-  int32 addition_stage = 4;
-  Time addition_time = 5;  // Reaction time
+  int32 addition_order = 4;
+  // When the addition event took place in terms of the reaction time (or, 
+  // in the case of flow chemistry, the residence time).
+  Time addition_time = 5;
   enum AdditionSpeed {
+    // Unspecified.
     ADDITION_SPEED_UNKNOWN = 0;
     ADDITION_SPEED_ALL_AT_ONCE = 1;
     ADDITION_SPEED_FAST = 2;
     ADDITION_SPEED_SLOW = 3;
     ADDITION_SPEED_DROPWISE = 4;
     ADDITION_SPEED_CONTINUOUS = 5;
   }
+  // The qualitative rate of addition.
   AdditionSpeed addition_speed = 6;
 }
 
 message Compound {
+  // Set of identifiers used to uniquely define this compound.
+  // Solutions or mixed compounds should use the NAME identifier
+  // and list all constituent compounds in the "components" field.
   repeated CompoundIdentifier identifiers = 1;
+  // Quantity of this compound used in the reaction.
   Amount amount = 2;
```

Comment:
Should amount/components be a Oneof?

Reply:
I envision a use-case where "amount" is used to specify the volume of a stock solution that is added, while "components" specifies the make-up of the solution, whose total volume can exceed the amount that was added in this particular reaction.